src/det.jl

```diff
@@ -7,25 +7,22 @@
     @inbounds return A[1]*A[4] - A[3]*A[2]
 end
 
-@inline function _det(::Size{(2,2)}, A::StaticMatrix{<:Any, <:Any, <:Unsigned})
-    @inbounds return Signed(A[1]*A[4]) - Signed(A[3]*A[2])
-end
-
 @inline function _det(::Size{(3,3)}, A::StaticMatrix)
     @inbounds x0 = SVector(A[1], A[2], A[3])
     @inbounds x1 = SVector(A[4], A[5], A[6])
     @inbounds x2 = SVector(A[7], A[8], A[9])
     return vecdot(x0, cross(x1, x2))
 end
 
-@inline function _det(::Size{(3,3)}, A::StaticMatrix{<:Any, <:Any, <:Unsigned})
-    @inbounds x0 = SVector(Signed(A[1]), Signed(A[2]), Signed(A[3]))
-    @inbounds x1 = SVector(Signed(A[4]), Signed(A[5]), Signed(A[6]))
-    @inbounds x2 = SVector(Signed(A[7]), Signed(A[8]), Signed(A[9]))
-    return vecdot(x0, cross(x1, x2))
+@generated function det(A::StaticMatrix{<:Any,<:Any,T}) where T <:Unsigned
+    T1 = promote_type(T,typeof(signed(one(T))))
+    (T1 <: Unsigned) && (T1 = Float64) #if still unsigned cast to Float64
```

Comment:
Yeah... not surprised about overflows as we've tended to optimize for speed rather than correctness in this package. Not sure if this is a good choice or not?